src/tests/math/test_math.f90

```diff
@@ -0,0 +1,129 @@
+! SPDX-Identifier: MIT
+
+module test_math
+    use stdlib_kinds, only: int8, int16, int32, int64, sp, dp, qp
+    use stdlib_error, only : check
+    use stdlib_math
+
+    implicit none
+
+contains
+
+    subroutine test_clip_int8(x, xmin, xmax, compare)
+        integer(int8), intent(in) :: x, xmin, xmax, compare
+
+        call check(clip(x, xmin, xmax) == compare)
+    
+    end subroutine test_clip_int8
+
+    subroutine test_clip_int16(x, xmin, xmax, compare)
+        integer(int16), intent(in) :: x, xmin, xmax, compare
+
+        call check(clip(x, xmin, xmax) == compare)
+    
+    end subroutine test_clip_int16
+
+    subroutine test_clip_int32(x, xmin, xmax, compare)
+        integer(int32), intent(in) :: x, xmin, xmax, compare
+
+        call check(clip(x, xmin, xmax) == compare)
+    
+    end subroutine test_clip_int32
+
+    subroutine test_clip_int64(x, xmin, xmax, compare)
+        integer(int64), intent(in) :: x, xmin, xmax, compare
+
+        call check(clip(x, xmin, xmax) == compare)
+    
+    end subroutine test_clip_int64
+
+    subroutine test_clip_sp(x, xmin, xmax, compare)
+        real(sp), intent(in) :: x, xmin, xmax, compare
+
+        call check(clip(x, xmin, xmax) == compare)
+    
+    end subroutine test_clip_sp
+
+    subroutine test_clip_dp(x, xmin, xmax, compare)
+        real(dp), intent(in) :: x, xmin, xmax, compare
+
+        call check(clip(x, xmin, xmax) == compare)
+    
+    end subroutine test_clip_dp
+
+    subroutine test_clip_qp(x, xmin, xmax, compare)
+        real(qp), intent(in) :: x, xmin, xmax, compare
+
+        call check(clip(x, xmin, xmax) == compare)
+    
+    end subroutine test_clip_qp
```

Comment:
Just a small nitpick. The variable `compare` could be renamed `answer` for clarity.